testflight/container_limits_test.go

```diff
@@ -16,7 +16,8 @@ var _ = Describe("A job with a task that has container limits", func() {
 		watch := spawnFly("trigger-job", "-j", inPipeline("container-limits-job"), "-w")
 		<-watch.Exited
 
-		if strings.Contains(string(watch.Out.Contents()), "memsw.limit_in_bytes: permission denied") {
+		// Guardian runtime will always fail this test unless it's explicitly told not to set the swap limit
+		if strings.Contains(string(watch.Out.Contents()), "memsw.limit_in_bytes: no such file or directory") {
```

Comment:
The Guardian runtime is failing because they're using an older version of runC with the `permission denied` error, maybe we should be doing regex matching instead? `failed to write "[0-9]+" to "/sys/fs/cgroup/memory/.*/memory.memsw.limit_in_bytes"`

The full error looks like
```
find or create container on worker fe03fcc43bdb: runc run: exit status 1: container_linux.go:367: starting container process caused: process_linux.go:459: container init caused: process_linux.go:422: setting cgroup config for procHooks process caused: failed to write "1073741824" to "/sys/fs/cgroup/memory/garden/b6ffb7e4-da29-462c-6514-3550aca2d4b6/memory.memsw.limit_in_bytes": open /sys/fs/cgroup/memory/garden/b6ffb7e4-da29-462c-6514-3550aca2d4b6/memory.memsw.limit_in_bytes: permission denied
```

Reply:
I'd actually prefer to let this fail since it will pass when bumping runc to rc-93. Will be a good indicator that we're using the latest version of runc. wdyt?

Reply:
I would argue that the test should not fail and be skipped regardless of the rc version. It's testing to see if the container limits are respected, if it's not possible to modify the cgroup, then this test shouldn't even run.

It's our fragile parsing of the error message that's causing it to fail instead of skip. The criteria for skipping should be "writing to the memory.limit_in_bytes cgroup failed (for whatever reason)", not "this file doesn't exist" or "you don't have permissions"


> Will be a good indicator that we're using the latest version of runc.

This might be very purist, but why should the test (or by extension us) care? We do not rely on any special features of rc93, so as long as it doesn't break our code, it should be treated like any other dependency and get bumped when it gets bumped.